modules/blockbuilder/blockbuilder.go

```diff
@@ -265,7 +265,10 @@ func (b *BlockBuilder) consume(ctx context.Context) (time.Duration, error) {
 
 	// First iteration over all the assigned partitions to get their current lag in time
 	for i, p := range ps {
-		if !p.hasRecords() { // No records, we can skip the partition
+		if !p.hasRecords() { // No records, skip for the first iteration
+			ps[i].lastRecordTs = time.Now()
```

Comment:
Let's leave a comment here like:  `We treat the partition as updated through now, and will check it again after ConsumeCycleDuration has elapsed`